test/rules/prefer_const_declarations.dart

```diff
@@ -32,3 +32,17 @@ class A {
 
   final i = const []; // OK
 }
+
+m() {
+  const o1 = const []; // OK
+  final o2 = []; // OK
+  final o3 = const []; // LINT
+  final o4 = ''; // LINT
+  final o5 = 1; // LINT
+  final o6 = 1.3; // LINT
+  final o7 = null; // LINT
+  final o8 = const {}; // LINT
```

Comment:
Can we add a `const` constructor to `A`, a method `int m() => 0`, and the following test cases?
```
final o9 = {};
final o10 = new A();
final o11 = const A();
final o12 = A.o1;
final o13 = A.o2;
final o14 = 011.m();
```

Reply:
Done.